solutions/usaco-639.mdx

```diff
@@ -23,59 +22,102 @@ author: Danh Ta Chi Thanh, Ryan Chou
 
 using namespace std;
 
-int main(){
-	ios_base::sync_with_stdio(0); cin.tie(0);
-	freopen("diamond.in","r",stdin);
-	freopen("diamond.out","w",stdout);
-	int n,k;
+int main() {
+	freopen("diamond.in", "r", stdin);
+	int n, k;
 	cin >> n >> k;
-	vector<int> a;
-	a.resize(n);
-	for (int i=0;i<n;++i) {
-		cin >> a[i];
+	vector<int> diamonds(n);
+	for (int& d : diamonds) {
+		cin >> d;
 	}
-	sort(a.begin(),a.end());
-	int ans=0;
-	for (int i=0;i<n;++i) {
-		int cnt=1;
-		for (int j=i+1;j<n;++j) {
-			if (a[j]-a[i]<=k) ++cnt;
-			else break;
+
+	int most = 0;
+	/*
+	* Iterate through all diamonds and test setting them
+	* as the smallest diamond in the case.
+	*/
+	for (int x : diamonds) {
```

Comment:
```suggestion
	for (int& x : diamonds) {
```

Reply:
i think the cost here is negligible since `int`s are primitive?